sdk/core/Azure.Core/src/AsyncCollection.cs

```diff
@@ -66,7 +66,16 @@ public abstract class AsyncCollection<T> : IAsyncEnumerable<Response<T>>
         /// enumerating asynchronously.
         /// </param>
         /// <returns>An async sequence of values.</returns>
-        public abstract IAsyncEnumerator<Response<T>> GetAsyncEnumerator(CancellationToken cancellationToken = default);
+        public virtual async IAsyncEnumerator<Response<T>> GetAsyncEnumerator(CancellationToken cancellationToken = default)
+        {
+            await foreach (Page<T> page in ByPage())
```

Comment:
Shouldn't we thread the cancellationToken into ByPage?  I had an internal GetNextPageAsync to handle this in https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Common/src/StorageAsyncCollection.cs